pkg/registry/apis/datasource/sub_health.go

```diff
@@ -0,0 +1,67 @@
+package datasource
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/grafana/grafana/pkg/apis/datasource/v0alpha1"
+)
+
+type subHealthREST struct {
+	builder *DataSourceAPIBuilder
+}
+
+var _ = rest.Connecter(&subHealthREST{})
+
+func (r *subHealthREST) New() runtime.Object {
+	return &v0alpha1.HealthCheckResult{}
+}
+
+func (r *subHealthREST) Destroy() {
+}
+
+func (r *subHealthREST) ConnectMethods() []string {
+	return []string{"GET"}
+}
+
+func (r *subHealthREST) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, ""
+}
+
+func (r *subHealthREST) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		pluginCtx, err := r.builder.getDataSourcePluginContext(ctx, name)
+		if err != nil {
+			responder.Error(err)
+			return
+		}
+
+		healthResponse, err := r.builder.client.CheckHealth(ctx, &backend.CheckHealthRequest{
+			PluginContext: *pluginCtx,
+		})
+		if err != nil {
+			responder.Error(err)
+			return
+		}
+
+		rsp := &v0alpha1.HealthCheckResult{}
+		rsp.Code = int(healthResponse.Status)
+		rsp.Status = healthResponse.Status.String()
+		rsp.Message = healthResponse.Message
+
+		if len(healthResponse.JSONDetails) > 0 {
+			err = json.Unmarshal(healthResponse.JSONDetails, &rsp.Details)
+			if err != nil {
+				responder.Error(err)
+				return
+			}
+		}
+
+		responder.Object(200, rsp) // TODO... 200 vs ????
```

Comment:
^^ @marefr -- what is the right response here?

Reply:
200 or 400 
https://github.com/grafana/grafana/blob/0b511aaace564461b1e117654eef1771c49cb3a7/pkg/api/datasources.go#L965-L969